velox/duckdb/conversion/DuckConversion.cpp

```diff
@@ -138,7 +138,41 @@ variant duckValueToVariant(const Value& val) {
     case LogicalTypeId::DOUBLE:
       return variant(val.GetValue<double>());
     case LogicalTypeId::DECIMAL:
-      return variant(val.GetValue<double>());
+      if (handleDecimalAsDouble) {
+        return variant(val.GetValue<double>());
+      } else {
+        uint8_t precision;
+        uint8_t scale;
+        int64_t unscaledValue;
+        val.type().GetDecimalProperties(precision, scale);
+        switch (val.type().InternalType()) {
+          case ::duckdb::PhysicalType::INT128: {
+            auto unscaledValue = val.GetValueUnsafe<::duckdb::hugeint_t>();
+            return variant::longDecimal(
+                buildInt128(unscaledValue.upper, unscaledValue.lower),
+                DECIMAL(precision, scale));
```

Comment:
DECIMAL(precision, scale) is repeated a couple of times. Consider creating a long variable. Also, consider extracting a help function to concert val.type() into DecimalType.